sunpy/net/jsoc/jsoc.py

```diff
@@ -228,6 +330,66 @@ def query(self, *query, **kwargs):
+        Builds a jsoc query, similar to query() method, and takes similar inputs.
+
+        Complex queries to be easily formed using logical operators such as
+        `&` and `|`, in the same way as the query() function.
+
+        *Example*
+
+        Request metadata or all all AIA 304 image data between 2014-01-01T00:00 and
+        2014-01-01T01:00.
+
+        Since, the function only performs a lookdata, and does not make a proper export
+        request, attributes like Segment need not be passed.
+
+        >>> import astropy.units as u
+        >>> from sunpy.net import jsoc
+        >>> from sunpy.net import attrs as a
+        >>> client = jsoc.JSOCClient()
+        >>> metadata = client.search_metadata(
+                                    a.jsoc.Time('2014-01-01T00:00:00', '2014-01-01T00:02:00'),
+        ...                         a.jsoc.Series('aia.lev1_euv_12s'), a.jsoc.Wavelength(304*u.AA))
+
+        Returns
+        -------
+        metadata : Pandas DataFrame
+            A collection of metadata of all the files.
+
+
+                                                          T_REC                  ...         T_REC_epoch
+        aia.lev1_euv_12s[2014-01-01T00:00:01Z][304]  2014-01-01T00:00:01Z        ...     1993.01.01_00:00:04_TAI
+        aia.lev1_euv_12s[2014-01-01T00:00:13Z][304]  2014-01-01T00:00:13Z        ...     1993.01.01_00:00:04_TAI
+        aia.lev1_euv_12s[2014-01-01T00:00:25Z][304]  2014-01-01T00:00:25Z        ...     1993.01.01_00:00:04_TAI
+        aia.lev1_euv_12s[2014-01-01T00:00:37Z][304]  2014-01-01T00:00:37Z        ...     1993.01.01_00:00:04_TAI
+        aia.lev1_euv_12s[2014-01-01T00:00:49Z][304]  2014-01-01T00:00:49Z        ...     1993.01.01_00:00:04_TAI
+        aia.lev1_euv_12s[2014-01-01T00:01:01Z][304]  2014-01-01T00:01:01Z        ...     1993.01.01_00:00:04_TAI
+        aia.lev1_euv_12s[2014-01-01T00:01:13Z][304]  2014-01-01T00:01:13Z        ...     1993.01.01_00:00:04_TAI
+        aia.lev1_euv_12s[2014-01-01T00:01:25Z][304]  2014-01-01T00:01:25Z        ...     1993.01.01_00:00:04_TAI
+        aia.lev1_euv_12s[2014-01-01T00:01:37Z][304]  2014-01-01T00:01:37Z        ...     1993.01.01_00:00:04_TAI
+        aia.lev1_euv_12s[2014-01-01T00:01:49Z][304]  2014-01-01T00:01:49Z        ...     1993.01.01_00:00:04_TAI
+        aia.lev1_euv_12s[2014-01-01T00:02:01Z][304]  2014-01-01T00:02:01Z        ...     1993.01.01_00:00:04_TAI
+
+
+        [11 rows x 176 columns]
+
+        """
+        query = and_(*query)
+        blocks = []
+        res = pd.DataFrame()
+        for block in walker.create(query):
+            iargs = kwargs.copy()
+            iargs.update(block)
+            iargs.update({'meta': True})
+            blocks.append(iargs)
+            res = res.append(self._lookup_records(iargs))
+
+        return res
+
     def request_data(self, jsoc_response, **kwargs):
         """
         Request that JSOC stages the data for download.
```

Comment:
Docstring tweaks to link  to classes in both parameters and returns.

Reply:
Done